xmodule/video_module/transcripts_utils.py

```diff
@@ -153,6 +154,68 @@ def youtube_video_transcript_name(youtube_text_api):
     return None
 
 
+def get_transcript_link_from_youtube(youtube_id):
+    """
+    Get the link for YouTube transcript by parsing the source of the YouTube webpage.
+    Inside the webpage, the details of the transcripts are located in a JSON object.
+    After prettifying the object, it looks like:
+
+    "captions": {
+        "playerCaptionsTracklistRenderer": {
+            "captionTracks": [
+                {
+                    "baseUrl": "...",
+                    "name": {
+                        "simpleText": "(Japanese in local language)"
+                    },
+                    "vssId": ".ja",
+                    "languageCode": "ja",
+                    "isTranslatable": true
+                },
+                {
+                    "baseUrl": "...",
+                    "name": {
+                        "simpleText": "(French in local language)"
+                    },
+                    "vssId": ".fr",
+                    "languageCode": "fr",
+                    "isTranslatable": true
+                },
+                {
+                    "baseUrl": "...",
+                    "name": {
+                        "simpleText": "(English in local language)"
+                    },
+                    "vssId": ".en",
+                    "languageCode": "en",
+                    "isTranslatable": true
+                },
+                ...
+            ],
+            "audioTracks": [...]
+            "translationLanguages": ...
+        },
+        ...
+    }
+
+    So we use a regex to find the captionTracks JavaScript array, and then convert it
+    to a Python dict and return the link for en caption
+    """
+    youtube_url_base = settings.YOUTUBE['TRANSCRIPTS']['YOUTUBE_URL_BASE']
+    try:
+        youtube_html = requests.get(f"{youtube_url_base}{youtube_id}")
+        caption_re = settings.YOUTUBE['TRANSCRIPTS']['CAPTION_TRACKS_REGEX']
+        caption_matched = caption_re.search(youtube_html.content.decode("utf-8"))
```

Comment:
This doesn't work:
```
cms_1                        | During handling of the above exception, another exception occurred:
cms_1                        | 
cms_1                        | Traceback (most recent call last):
cms_1                        |   File "/openedx/venv/lib/python3.8/site-packages/django/core/handlers/exception.py", line 47, in inner
cms_1                        |     response = get_response(request)
cms_1                        |   File "/openedx/venv/lib/python3.8/site-packages/django/core/handlers/base.py", line 181, in _get_response
cms_1                        |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
cms_1                        |   File "/opt/pyenv/versions/3.8.12/lib/python3.8/contextlib.py", line 75, in inner
cms_1                        |     return func(*args, **kwds)
cms_1                        |   File "/openedx/venv/lib/python3.8/site-packages/django/contrib/auth/decorators.py", line 21, in _wrapped_view
cms_1                        |     return view_func(request, *args, **kwargs)
cms_1                        |   File "/openedx/edx-platform/cms/djangoapps/contentstore/views/transcripts_ajax.py", line 341, in check_transcripts
cms_1                        |     if get_transcript_link_from_youtube(youtube_id):
cms_1                        |   File "/openedx/edx-platform/xmodule/video_module/transcripts_utils.py", line 208, in get_transcript_link_from_youtube
cms_1                        |     caption_matched = caption_re.search(youtube_html.content.decode("utf-8"))
cms_1                        | AttributeError: 'str' object has no attribute 'search'
```
Please try:
```suggestion
        caption_matched = re.search(caption_re, youtube_html.content.decode("utf-8"))
```

Reply:
ohh. I forgot to fix that... seems what returned from settings is a string instead of a regex object.